shape_completion_training/src/shape_completion_training/model/filepath_tools.py

```diff
@@ -17,79 +18,65 @@ def unique_trial_name(*names):
-    The directory is created if it does not exist.
-
-    Call as: get_trial_directory("/path/to/trials/directory/")
+    Returns the filepath to the directory for the trial
     """
+    r = rospkg.RosPack()
+    shape_completion_training_path = r.get_path('shape_completion_training')
+    base_directory = join(shape_completion_training_path, 'trials')
     if not os.path.isdir(base_directory):
         _make_new_trials_directory(base_directory)
 
     if nick is None:
         nick = _make_tmp_nick(base_directory)
 
     unique_trial_subdirectory_name = unique_trial_name()
-    fp = join(base_directory, nick, unique_trial_subdirectory_name)
-    os.mkdir(fp)
+    fp = join(base_directory, nick)
+    if not os.path.exists(fp):
+        os.mkdir(fp)
+    full_trial_directory = join(base_directory, nick, unique_trial_subdirectory_name)
+    if os.path.exists(full_trial_directory):
+        raise RuntimeError("full trial directory already exists!?")
+    os.mkdir(full_trial_directory)
 
     if write_summary:
-        _write_summary(fp, nick)
+        _write_summary(full_trial_directory, nick)
 
-    print("Running trial {} at {}".format(nick, fp))
-    return fp
+    print("Running trial {} at {}".format(nick, full_trial_directory))
+    return full_trial_directory
 
 
-def handle_params(default_params_fp, model_params_fp, given_params):
+def handle_params(trial_directory, given_params):
     """
     Handles loading and saving of the params
     If params is None it will load from params from the directory
     Otherwise it save params to the directory
     Returns the params loaded or saved
-    Prompts user if default parameters do not match the given parameters
-
-    TODO: If given_params is specified this will silently overwrite any params already part of the model.
     """
-    if given_params is None:
-        return _load_params(default_params_fp, model_params_fp)
-
-    ### Check defaults
-    with open(join(default_params_fp, 'default_params.json'), 'r') as f:
-        default_params = json.load(f)
-
-    given_keys = set(given_params.keys())
-    default_keys = set(default_params.keys())
-
-    if given_keys != default_keys:
-        print()
-        print("!! Warning !!")
-        print("Default params and given params have different keys. The defaults should have the same entry names as "
-              "the specified params. This difference may prevent properly reloading after future changes!")
-        for missing_default_key in given_keys - default_keys:
-            print("{} missing from defaults".format(missing_default_key))
-        print()
-        for missing_given_key in default_keys - given_keys:
-            print("{} missing from specified keys".format(missing_given_key))
-        print("Press any key to continue")
-        raw_input()
+    model_params = json.load(open(join(trial_directory, "params.json"), 'r'))
+    return _handle_params(model_params, given_params)
 
-    _write_params(model_params_fp, given_params)
```

Comment:
Where do you handle writing the params file now?
Some params file needs to be written for new trials